play-json/jvm/src/test/scala/play/api/libs/json/JsonSpec.scala

```diff
@@ -477,6 +477,19 @@ class JsonSpec extends org.specs2.mutable.Specification {
       fromJson[JsonNode](jsNum).map(_.toString).must_==(JsSuccess("12.345"))
     }
 
+    "Serialize JsNumbers with e notation correctly" in {
+      val numStrings = Seq(
+        "1.23456789012345679012345679e999",
+        "-1.23456789012345679012345679e999",
+        "1.23456789012345679012345679e-999",
+        "-1.23456789012345679012345679e-999"
```

Comment:
Not only "positive" test about decimal case must be covered, but the other non decimal case

Reply:
more test cases added - also to #1077 to show that the cases already pass without this set of changes